types/plotly.js/index.d.ts

```diff
@@ -21,6 +21,8 @@
 //                 Oleg Shilov <https://github.com/olegshilov>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+// Minimum TypeScript Version: 3.4
```

Comment:
So this one is problematic IMO,
it works here (in DT) as all packages referencing this one are compatible with 3.4,
but this could break code in the wild,
what about redefining the `marker` to fit new shape in extension:

```diff
diff --git a/types/plotly.js/index.d.ts b/types/plotly.js/index.d.ts
index b3b9ef1649..bf5b83b4a1 100644
--- a/types/plotly.js/index.d.ts
+++ b/types/plotly.js/index.d.ts
@@ -21,8 +21,6 @@
 //                 Oleg Shilov <https://github.com/olegshilov>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

-// Minimum TypeScript Version: 3.4
-
 import * as _d3 from 'd3';
 import { BoxPlotData } from './lib/traces/box';
 import { ViolinData } from './lib/traces/violin';
diff --git a/types/plotly.js/lib/traces/box.d.ts b/types/plotly.js/lib/traces/box.d.ts
index 441ce57ddf..a631c93112 100644
--- a/types/plotly.js/lib/traces/box.d.ts
+++ b/types/plotly.js/lib/traces/box.d.ts
@@ -1,8 +1,15 @@
 import { PlotData, Color, MarkerSymbol } from '../../index';
 import { ScatterSelectedMarker } from './scatter';

+// Remove redefinition of Omit after TS minimum version progresses to 3.4
+/**
+ * Construct a type with the properties of T except for those in type K.
+ */
+export type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
+
 // See https://github.com/plotly/plotly.js/blob/master/src/traces/box/attributes.js
-export interface BoxPlotData extends PlotData {
+// remove Omit after TS minimum version progresses to 3.4
+export interface BoxPlotData extends Omit<PlotData, 'marker'> {
```
that would remove the DT increment requirement, should pass a DT tests, not sure if that is a 
best solution though,
thoughts?

Reply:
I'm not sure about this one either. How about the changes in c2b9bde below? This removes the Typescript version issue and all tests pass.